`test_huber_loss` is flaky

This is part of https://github.com/chainer/chainer/issues/6903 effort.

- Failure example: https://ci.preferred.jp/chainer.py27and35/2312/
- Failure test: chainer_tests/functions_tests/loss_tests/test_huber_loss.py
- Failure test case: `TestHuberLoss_param_14_{double_backward_options={'eps': 0.01, 'atol': 0.1, 'rtol': 0.1}, forward_options={}, reduce='no', backward_options={'eps': 0.01, 'atol': 0.01, 'rtol': 0.01}, shape=(4, 10), dtype=float64}.test_double_backward_gpu`
- Tolerance:
```
E           Not equal to tolerance rtol=0.1, atol=0.1
E           
E           Mismatch: 100%
E           Max absolute difference: 0.13130243
E           Max relative difference: 1.99520297
```

--- a/tests/chainer_tests/functions_tests/loss_tests/test_huber_loss.py
+++ b/tests/chainer_tests/functions_tests/loss_tests/test_huber_loss.py
@@ -99,7 +99,7 @@ class TestHuberLoss(unittest.TestCase):
 
         delta = 1
         eps = self.double_backward_options['eps']
-        xp = cuda.cupy.get_array_module(x_data)
+        xp = chainer.get_array_module(x_data)
         mask = xp.abs(xp.abs(x_data - t_data) - delta) < eps
         x_data[mask] = 0
         t_data[mask] = 0

--- a/tests/chainer_tests/functions_tests/loss_tests/test_huber_loss.py
+++ b/tests/chainer_tests/functions_tests/loss_tests/test_huber_loss.py
@@ -18,13 +18,13 @@ from chainer import utils
       'double_backward_options': {'eps': 1e-1, 'rtol': 1e0, 'atol': 1e0}},
      {'dtype': numpy.float32,
       'forward_options': {},
-      'backward_options': {'eps': 1e-2, 'rtol': 1e-2, 'atol': 1e-2},
-      'double_backward_options': {'eps': 1e-2, 'rtol': 1e-1, 'atol': 1e-1}},
+      'backward_options': {'eps': 1e-3, 'rtol': 1e-2, 'atol': 1e-2},
+      'double_backward_options': {'eps': 1e-3, 'rtol': 1e-2, 'atol': 1e-2}},
      {'dtype': numpy.float64,
       'forward_options': {},
-      'backward_options': {'eps': 1e-2, 'rtol': 1e-2, 'atol': 1e-2},
-      'double_backward_options': {'eps': 1e-2, 'rtol': 1e-1, 'atol': 1e-1}},
-     ],
+      'backward_options': {'eps': 1e-3, 'rtol': 1e-2, 'atol': 1e-2},
+      'double_backward_options': {'eps': 1e-3, 'rtol': 1e-2, 'atol': 1e-2}},
+    ],
     testing.product({
         'shape': [(), (3,)],
         'reduce': ['no'],
@@ -96,8 +96,16 @@ class TestHuberLoss(unittest.TestCase):
 
     def check_double_backward(self, x_data, t_data, y_grad, x_grad_grad,
                               t_grad_grad):
+
+        delta = 1
+        eps = self.double_backward_options['eps']
+        xp = cuda.cupy.get_array_module(x_data)
+        mask = xp.abs(xp.abs(x_data - t_data) - delta) < eps
+        x_data[mask] = 0
+        t_data[mask] = 0
+
         def f(x, t):
-            return functions.huber_loss(x, t, delta=1, reduce=self.reduce)
+            return functions.huber_loss(x, t, delta=delta, reduce=self.reduce)
 
         gradient_check.check_double_backward(
             f, (x_data, t_data), y_grad, (x_grad_grad, t_grad_grad),

--- a/tests/chainer_tests/functions_tests/loss_tests/test_huber_loss.py
+++ b/tests/chainer_tests/functions_tests/loss_tests/test_huber_loss.py
@@ -99,7 +99,7 @@ class TestHuberLoss(unittest.TestCase):
 
         delta = 1
         eps = self.double_backward_options['eps']
-        xp = chainer.get_array_module(x_data)
+        xp = chainer.backend.get_array_module(x_data)
         mask = xp.abs(xp.abs(x_data - t_data) - delta) < eps
         x_data[mask] = 0
         t_data[mask] = 0

--- a/tests/chainer_tests/functions_tests/loss_tests/test_huber_loss.py
+++ b/tests/chainer_tests/functions_tests/loss_tests/test_huber_loss.py
@@ -14,8 +14,8 @@ from chainer import utils
 @testing.parameterize(*testing.product_dict(
     [{'dtype': numpy.float16,
       'forward_options': {'rtol': 5e-3, 'atol': 5e-3},
-      'backward_options': {'eps': 1e-1, 'rtol': 1e0, 'atol': 1e0},
-      'double_backward_options': {'eps': 1e-1, 'rtol': 1e0, 'atol': 1e0}},
+      'backward_options': {'eps': 1e-1, 'rtol': 1e-1, 'atol': 1e-1},
+      'double_backward_options': {'eps': 1e-1, 'rtol': 1e-1, 'atol': 1e-1}},
      {'dtype': numpy.float32,
       'forward_options': {},
       'backward_options': {'eps': 1e-3, 'rtol': 1e-2, 'atol': 1e-2},
@@ -24,7 +24,7 @@ from chainer import utils
       'forward_options': {},
       'backward_options': {'eps': 1e-3, 'rtol': 1e-2, 'atol': 1e-2},
       'double_backward_options': {'eps': 1e-3, 'rtol': 1e-2, 'atol': 1e-2}},
-    ],
+     ],
     testing.product({
         'shape': [(), (3,)],
         'reduce': ['no'],
@@ -40,7 +40,7 @@ class TestHuberLoss(unittest.TestCase):
         self._config_user.__enter__()
 
         self.x = utils.force_array(
-            (numpy.random.random(self.shape) - 0.5) * 20, self.dtype)
+            (numpy.random.random(self.shape) - 0.5) * 4, self.dtype)
         self.t = utils.force_array(numpy.random.random(self.shape), self.dtype)
         if self.reduce == 'sum_along_second_axis':
             gy_shape = self.shape[:1] + self.shape[2:]

